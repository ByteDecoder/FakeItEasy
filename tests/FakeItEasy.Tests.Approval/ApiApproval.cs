namespace FakeItEasy.Tests.Approval
{
    using System;
    using System.Diagnostics.CodeAnalysis;
    using System.IO;
    using System.Reflection;
    using System.Runtime.CompilerServices;
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using PublicApiGenerator;
    using Xunit;

    public class ApiApproval
    {
        [Fact]
        [UseReporter(typeof(DiffReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveApi45()
        {
            ApproveApi("net45");
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveApi40()
        {
            ApproveApi("net40");
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveApiNetStd16()
        {
            ApproveApi("netstandard1.6");
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveApiNetStd20()
        {
            ApproveApi("netstandard2.0");
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveApiNetStd21()
        {
            ApproveApi("netstandard2.1");
        }

        [MethodImpl(MethodImplOptions.NoInlining)]
        private static void ApproveApi(string frameworkVersion)
        {
            string codeBase = Assembly.GetExecutingAssembly().CodeBase;
            UriBuilder uri = new UriBuilder(new Uri(codeBase));
            string assemblyPath = Uri.UnescapeDataString(uri.Path);
            var containingDirectory = Path.GetDirectoryName(assemblyPath);
            var configurationName = new DirectoryInfo(containingDirectory).Parent.Name;
            var assemblyFile = $@"..\..\..\..\..\src\FakeItEasy\bin\{configurationName}\{frameworkVersion}\FakeItEasy.dll";

            var assembly = Assembly.LoadFile(Path.GetFullPath(assemblyFile));
            Approvals.Verify(ApiGenerator.GeneratePublicApi(assembly));
        }
    }
}
