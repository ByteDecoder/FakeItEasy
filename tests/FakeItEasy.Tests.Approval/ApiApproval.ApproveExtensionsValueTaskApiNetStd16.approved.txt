[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETStandard,Version=v1.6", FrameworkDisplayName="")]
namespace FakeItEasy
{
    public class static ValueTaskReturnValueConfigurationExtensions
    {
        public static FakeItEasy.Configuration.IAfterCallConfiguredWithOutAndRefParametersConfiguration<FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<T>>> Returns<T>(this FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<T>> configuration, T value) { }
        public static FakeItEasy.Configuration.IAfterCallConfiguredWithOutAndRefParametersConfiguration<FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<T>>> ReturnsLazily<T>(this FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<T>> configuration, System.Func<T> valueProducer) { }
        public static FakeItEasy.Configuration.IAfterCallConfiguredWithOutAndRefParametersConfiguration<FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>>> ReturnsLazily<TReturnType, T1>(this FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>> configuration, System.Func<T1, TReturnType> valueProducer) { }
        public static FakeItEasy.Configuration.IAfterCallConfiguredWithOutAndRefParametersConfiguration<FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>>> ReturnsLazily<TReturnType, T1, T2>(this FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>> configuration, System.Func<T1, T2, TReturnType> valueProducer) { }
        public static FakeItEasy.Configuration.IAfterCallConfiguredWithOutAndRefParametersConfiguration<FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>>> ReturnsLazily<TReturnType, T1, T2, T3>(this FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>> configuration, System.Func<T1, T2, T3, TReturnType> valueProducer) { }
        public static FakeItEasy.Configuration.IAfterCallConfiguredWithOutAndRefParametersConfiguration<FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>>> ReturnsLazily<TReturnType, T1, T2, T3, T4>(this FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>> configuration, System.Func<T1, T2, T3, T4, TReturnType> valueProducer) { }
        public static FakeItEasy.Configuration.IAfterCallConfiguredWithOutAndRefParametersConfiguration<FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>>> ReturnsLazily<TReturnType, T1, T2, T3, T4, T5>(this FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>> configuration, System.Func<T1, T2, T3, T4, T5, TReturnType> valueProducer) { }
        public static FakeItEasy.Configuration.IAfterCallConfiguredWithOutAndRefParametersConfiguration<FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>>> ReturnsLazily<TReturnType, T1, T2, T3, T4, T5, T6>(this FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>> configuration, System.Func<T1, T2, T3, T4, T5, T6, TReturnType> valueProducer) { }
        public static FakeItEasy.Configuration.IAfterCallConfiguredWithOutAndRefParametersConfiguration<FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>>> ReturnsLazily<TReturnType, T1, T2, T3, T4, T5, T6, T7>(this FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>> configuration, System.Func<T1, T2, T3, T4, T5, T6, T7, TReturnType> valueProducer) { }
        public static FakeItEasy.Configuration.IAfterCallConfiguredWithOutAndRefParametersConfiguration<FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>>> ReturnsLazily<TReturnType, T1, T2, T3, T4, T5, T6, T7, T8>(this FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<TReturnType>> configuration, System.Func<T1, T2, T3, T4, T5, T6, T7, T8, TReturnType> valueProducer) { }
        public static void ReturnsNextFromSequence<T>(this FakeItEasy.Configuration.IReturnValueConfiguration<System.Threading.Tasks.ValueTask<T>> configuration, params T[] values) { }
    }
}